### PR TITLE
Fix PWM for esp32

### PR DESCRIPTION
**Pull request by alberk8**
## Description

- Fix the PWM not output and reporting the DutyCycle correctly.

## Motivation and Context

- Fixes/https://github.com/nanoframework/Home/issues/1334

## How Has This Been Tested?

ESP32 and ESP32 S3
Tested with this [Code](https://github.com/alberk8/nanoFramework-Pwm) 
## Screenshots
image

image

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by alberk8** [comment]
Additional Tests with [Repo](https://github.com/alberk8/nanoFramework-Pwm)

image

image

File: targets/ESP32/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_System_Device_Pwm_PwmChannel.cpp [hunk 1]
```diff
@@ -149,10 +151,26 @@ HRESULT sys_dev_pwm_native_System_Device_Pwm_PwmChannelHelpers::ConfigureAndStar
...
+    result = ledc_timer_config(&timer_conf);
+
+    if (result != ESP_OK)
+    {
+        NANOCLR_SET_AND_LEAVE(CLR_E_INVALID_PARAMETER);
+    }
+
     // Work out the duty Cycle for the current duty resolution
-    dutyCycle = CalculateDuty(timerId, dutyCycle, polarity);
+    dutyCycle = CalculateDuty(timerId, dutyCycle); // , polarity);
```
Comment (Ellerbach) [inline targets/ESP32/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_System_Device_Pwm_PwmChannel.cpp:171]: if you are not using polarity anymore, just remove it then, so no dead code

**Review by Ellerbach — APPROVED** [review]
A lot of work to have this fully cleaned up and working. Thanks a lot for the deep testing as well. Will that fix it for all the device family?

File: targets/ESP32/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_System_Device_Pwm_PwmChannel.cpp (hunk 1 above)
Comment (alberk8) [reply targets/ESP32/_nanoCLR/System.Device.Pwm/sys_dev_pwm_native_System_Device_Pwm_PwmChannel.cpp]: I have tested in ESP32 and ESP32 S3.

**Review by MrCSharp22 — APPROVED** [review]

**Pull request merged**